docs/halibut-protocol.md

```diff
@@ -0,0 +1,66 @@
+# Halibut Protocol data exchange
+
+This document outlines the data that is exchanged between the **Client** and **Service**. In this document:
+
+ - **Client** is what makes the remote procedure calls e.g. Octopus Server.
+ - **Service** is what executes the methods e.g. Tentacle. 
+
+
+When invoking a remote method in both polling and listening mode, the same general process is followed:
+
+1.  After the TCP connection is intiated, both server and client identify themselves to each other through a set of control messages.
+2. The Client sends a `Request` message containing the method to execute and data.
+3. The Service executes the method and sends the result in a `Response` message.
+4. The Service sends a `NEXT` control message, signaling it is ready for the next action.
+5. The Client sends a `Proceed` control message, signaling to the Service to be ready for another `Request`.
```

Comment:
The casing is different here compared to `NEXT` and `END`, is that how Halibut works?

Reply:
nice catch